clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji Sure. I have updated the test to verify that if we keep doing fetchRecords(), the code will return good records before the bad one, throw exception, and return good records after the bad one.

I think `PartitionRecords.nextFetchOffset` is not updated if there is an exception from parsing or validating any record.